src/nanovg_pick.c

```diff
@@ -0,0 +1,1651 @@
+						NVGpickPath** picked = realloc(ps->picked, sizeof(NVGpickPath*) * ps->cpicked);
+						if (picked == NULL)
+							break;
+						ps->cpicked = cpicked;
+						ps->picked = picked;
+					}
+					ps->picked[npicked] = pp;
+					++npicked;
+				}
+			}				
+			pp = pp->next;
+		}
+		
+		cellx >>= 1;
+		celly >>= 1;
+		levelwidth >>= 1;
+	}
+	
+	qsort(ps->picked, npicked, sizeof(NVGpickPath*), nvg__comparePaths);
+	
+	maxids = nvg__mini(maxids, npicked);
+	for (id = 0; id < maxids; ++id)
+		ids[id] = ps->picked[id]->id;
+		
+	return maxids;
+}
+
+// Returns the id of the pickable shape containing x,y or -1 if not shape is found.
+int nvgHitTest(NVGcontext* ctx, float x, float y, int flags)
+{
+	if (ctx->pickScene == NULL)	
+		return -1;
+	NVGpickScene* ps = ctx->pickScene;
+
+	int levelwidth = 1 << (ps->nlevels - 1);
+	int cellx = nvg__clampi((int)(x / ps->xdim), 0, levelwidth-1);
+	int celly = nvg__clampi((int)(y / ps->ydim), 0, levelwidth-1);
+
+	int bestOrder = -1;
+	int bestID = -1;
+	int hit = 0;
+	int lvl;	
+
+	for (lvl = ps->nlevels - 1; lvl >= 0; --lvl) {
+		NVGpickPath* pp = ps->levels[lvl][celly*levelwidth + cellx];
+		while(pp) {
+			if (nvg__pickPathTestBounds(ps, pp, x, y)) {		
+				
+				if ((flags & NVG_TEST_STROKE) && (pp->flags & NVG_PICK_STROKE))
+					hit = nvg__pickPathStroke(ps, pp, x, y);
+
+				if (!hit && (flags & NVG_TEST_FILL) && (pp->flags & NVG_PICK_FILL))
+					hit = nvg__pickPath(ps, pp, x, y);
+					
+				if (hit) {
+					if (pp->order > bestOrder) {
+						bestOrder = pp->order;
+						bestID = pp->id;
+					}
+				}
+			}				
+			pp = pp->next;
+		}
+		
+		cellx >>= 1;
+		celly >>= 1;
+		levelwidth >>= 1;		
+	}
+	
+	return bestID;
+}
+
+int nvgInFill(NVGcontext* ctx, float x, float y)
+{
+	NVGpickScene* ps = nvg__pickSceneGet(ctx);
+
+	int oldnpoints = ps->npoints;
+	int oldnsegments = ps->nsegments;	
+
+	NVGpickPath* pp = nvg__pickPathCreate(ctx, 1, 0);
+
+	if (x < pp->bounds[0] ||
+		y < pp->bounds[1] ||
+		x > pp->bounds[2] ||
+		y > pp->bounds[3])
+		return 0;
+	
+	int	hit = nvg__pickPath(ps, pp, x, y);
+
+	nvg__returnPickPath(ps, pp);
+
+	ps->npoints = oldnpoints;
+	ps->nsegments = oldnsegments;
+		
+	return hit;		
+}
+
+int nvgInStroke(NVGcontext* ctx, float x, float y)
+{
+	NVGpickScene* ps = nvg__pickSceneGet(ctx);
+
+	int oldnpoints = ps->npoints;
+	int oldnsegments = ps->nsegments;	
+
+	NVGpickPath* pp = nvg__pickPathCreate(ctx, 1, 1);
+
+	if (x < pp->bounds[0] ||
+		y < pp->bounds[1] ||
+		x > pp->bounds[2] ||
+		y > pp->bounds[3])
+		return 0;
+	
+	int	hit = nvg__pickPathStroke(ps, pp, x, y);
+
+	nvg__returnPickPath(ps, pp);
+
+	ps->npoints = oldnpoints;
+	ps->nsegments = oldnsegments;
+		
+	return hit;		
+}
+
+
+static int nvg__countBitsUsed(int v)
+{
+	if (v == 0)
+		return 0;
+#if defined(__clang__) || defined(__GNUC__)
+	return (sizeof(int) * 8) - __builtin_clz(v);
+#elif defined(_MSC_VER)
+	return (sizeof(int) * 8) -  __lzcnt(v);
+#else
+	// Poor mans 16bit clz variant
+	int test;
```

Comment:
It seems mighty odd to me to introduce a variable across all code variations here - all it will do is trigger "unused var" warnings when compiling on clang, gcc and MSVC when that warning is enabled. Unless you mean also protect it by #if guards as well (which seems to go against compactness).

Sure?